internal/resource/resource_plugin.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package resource
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/nginx/agent/v3/api/grpc/mpi/v1"
+
+	"github.com/nginx/agent/v3/internal/bus"
+)
+
+type Resource struct {
+	messagePipe     bus.MessagePipeInterface
+	resourceService resourceServiceInterface
+	resource        *v1.Resource
```

Comment:
I don't think the plugin should hold the `v1.Resource` in its state. It could just ask for it via the service's getter whenever it's needed.

Reply:
Ya I'll make that change